### PR TITLE
Wallet enhancements: send wallet <> payins/logs + passphrase improvements

### PR DESCRIPTION
**Pull request by huumn**
This does one too many things (and I should probably break out the payin changes ... commits ended up bleeding a bit into each other):

1. associates sending wallets with payins and wallet logs
    - this affords for richer wallet activity views and bidirectional references
    - associated wallet and wallet logs appear on transaction views now
2. gates wallet pages with the passphrase initialization/input on initial/subsequent devices
    - folks were consistently confused by passphrase prompting on new devices/browsers
    - while this change increases initial device friction, it is super clear that a passphrase is required
4. disambiguates passphrase copy, e.g. "stacker news wallet passphrase" and repeats itself several times
5. attempts to improve client/server key synchronization edge cases by sharing more code and addressing failures on either end

I wrote tons of ephemeral tests around this and QA is strong. This still needs:

- [ ] scrutinize/reason about changes more before merging
- [ ] fix human readability regressions
- [ ] pick out payin changes (should be possible)
- [ ] QA clientside migration with special attention to UX

Screenshot 2026-04-07 at 2 14 02 PM
Screenshot 2026-04-07 at 2 14 25 PM
Screenshot 2026-04-07 at 2 14 36 PM

---

> [!NOTE]
> **High Risk**
> Touches payment initiation/retry logic and wallet key/passphrase flow, which are security- and money-adjacent; mistakes could misattribute protocol usage, affect retry behavior, or lock users out of wallets.
> 
> **Overview**
> **Adds first-class `sendProtocolId` propagation across pay-ins and retries.** GraphQL mutations now accept `sendProtocolId`, the server validates/normalizes it, persists it onto `payInBolt11`, and uses it during `retry` (including nulling it for QR/manual fallback) and P2P zap selection.
> 
> **Improves pay-in/wallet observability.** Pay-in transitions now emit wallet logs that update protocol status (including settled/failed/canceled outcomes) and the PayIn view shows the associated wallet plus filtered `WalletLogs` for that `payInId`; the `walletLogs` query/cache keying was expanded to include `payInId`.
> 
> **Refactors wallet UX and key handling.** Wallet pages are gated through new route/page shells that enforce passphrase unlock/setup, passphrase UI was rewritten to support save/regenerate/reset flows, IndexedDB vault storage gained delete support, and wallet logging was consolidated into the payment hook with separate in-form (template) logs vs server logs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ceb46663f99a8bc8fc3c3a801dde0f71c3f4362b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: api/payIn/transitions.js [hunk 1]
```diff
@@ -553,8 +590,13 @@ export async function payInCancel ({ data, models, lnd, boss, ...args }) {
     if (transitionedPayIn.payOutBolt11) {
       const { protocolId, userId, bolt11 } = transitionedPayIn.payOutBolt11
       const logger = walletLogger({ protocolId, userId, payInId, models })
-      const decoded = await parsePaymentRequest({ request: bolt11 })
-      logger.info(`invoice for ${formatSats(msatsToSats(decoded.mtokens))} canceled by payer`)
+      try {
+        const decoded = parsePaymentRequest({ request: bolt11 })
```
Comment (sir-opti) [inline api/payIn/transitions.js:594]: lost `await`?

File: components/payIn/hooks/auto-retry-utils.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    await waitForWalletPayment(newPayIn)
+  } catch (err) {
+    if (isStopped()) {
+      // Stop/pause events should not strand the new attempt in a pending state.
+      await payInHelper.cancel(newPayIn).catch(() => {})
+      return
+    }
+    if (err instanceof WalletConfigurationError) {
+      // consume attempt by canceling invoice
+      await payInHelper.cancel(newPayIn)
```
Comment (sir-opti) [inline components/payIn/hooks/auto-retry-utils.js:52]: Note: this can throw (all other calls to `.cancel()` add `.catch()` - not sure if that is intended?

File: wallets/server/resolvers/wallet.js [hunk 3]
```diff
@@ -52,6 +51,53 @@ export const resolvers = {
   }
 }
 
+async function getVaultMetadata (models, userId) {
```
Comment (sir-opti) [inline wallets/server/resolvers/wallet.js:54]: probably more useful to import this from `./protocol.js` than to maintain it in 2 files.

**Review by sir-opti — COMMENTED** [review]
Have 3 minor code review findings pre-test

File: api/payIn/transitions.js (hunk 1 above)
Comment (huumn) [reply api/payIn/transitions.js:594]: This doesn't return a promise according to the docs: https://github.com/alexbosworth/ln-service?tab=readme-ov-file#parsepaymentrequest

Worth checking the code though.

File: components/payIn/hooks/auto-retry-utils.js (hunk 2 above)
Comment (huumn) [reply components/payIn/hooks/auto-retry-utils.js:52]: Ah good catch!

File: api/payIn/transitions.js (hunk 1 above)
Comment (sir-opti) [reply api/payIn/transitions.js:594]: found the [source](https://github.com/alexbosworth/invoices/blob/d43b4150d1266a929f93815edebc2ba9ee29f6cb/bolt11/parse_payment_request.js#L63) - it is indeed not async.

**Comment by huumn** [comment]
closing for #2909 and #2910
